### PR TITLE
Feature/#40 move business logic from controllers to a business layer

### PR DESCRIPTION
**Pull request by Thomsen96**
This pull request should resolve issue #40

In this have @ValarMarkhulis  and @Thomsen96  Moved all the business logic into a new package called Logic.

So that now in the controller classes are only the logic that translates the information from the presentation layer to a logic layer. 

And we also have moved all the communication to the database in a database package. 

So recap: **Make clear disinction between Presentation, Logic and Database layers.**

### CODE REVIEW

File: src/logic/user.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+		return models.User{}, errors.New("You have to enter a username")
+	} else if registationUser.Email == "" || !strings.Contains(registationUser.Email, "@") {
+		return models.User{}, errors.New("Your have to enter a valid email address")
+	} else if registationUser.Password1 == "" {
+		return models.User{}, errors.New("You have to enter a password")
+	} else if registationUser.Password1 != registationUser.Password2 {
+		return models.User{}, errors.New("The two passwords do not match")
+	} else {
+
+		user, err := database.GetUserFromDb(registationUser.Username)
```
Comment (Turboen) [inline src/logic/user.go:23]: _, err := database.GetUserFromDb(registationUser.Username)
		if err != nil && err.Error() == "User not found" { // TODO: This is bad error handling - should maybe be changed
			database.AddUserToDb(registationUser)
			return models.User{}, nil
		} else {
			return models.User{}, errors.New("The username is already taken")
		}

File: src/database/sqlite3.go [hunk 2]
```diff
@@ -190,6 +190,17 @@ func GetUserFromDb(username string) (models.User, error) {
 
 }
 
+func CheckIfUserExists(username string) bool {
```
Comment (Zavir) [inline src/database/sqlite3.go:193]: `
func CheckIfUserExists(username string) bool {

	user, _ := GetUserFromDb(username)

	if user.User_id != 0 {
		return true
	}

	return false
}
`

redundant else ;)

**Review by Zavir — APPROVED** [review]
Looks clean :), added one suggestion

File: src/database/sqlite3.go [hunk 3]
```diff
@@ -94,57 +96,164 @@ func GetAllMessages() []models.Message {
...
+
+	user, _ := GetUserFromDb(username)
+
+	if user.User_id != 0 {
+		return true
+	} else {
+		return false
 	}
+}
```
Comment (Thomsen96) [inline src/database/sqlite3.go:202]:
```suggestion
func CheckIfUserExists(username string) bool {

	user, _ := GetUserFromDb(username)
	if user.User_id != 0 {
		return true
	}

	return false
}
```

redundant else ;)

**Comment by Zavir** [comment]
Good job guys! Everything looks good :)

**Pull request merged**